test/e2e/dataplane/tcp_pod_connectivity.go

```diff
@@ -0,0 +1,115 @@
+package dataplane
+
+import (
+	"strings"
+
+	"github.com/submariner-io/submariner/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("[dataplane] Basic connectivity tests across clusters without discovery", func() {
+	f := framework.NewDefaultFramework("dataplane-conn-nd")
+	var useService bool
+
+	verifyInteraction := func(listenerScheduling, connectorScheduling framework.NetworkPodScheduling) {
+		It("should have sent the expected data from the pod to the other pod", func() {
+			listenerPod, connectorPod := runConnectivityTest(f, useService, listenerScheduling, connectorScheduling)
+			Expect(listenerPod.TerminationMessage).To(ContainSubstring(connectorPod.Config.Data))
+			Expect(connectorPod.TerminationMessage).To(ContainSubstring(listenerPod.Config.Data))
+
+			framework.Logf("Connector pod has IP: %s", connectorPod.Pod.Status.PodIP)
+			By("Verifying the output of listener pod which must contain the source IP")
+			Expect(listenerPod.TerminationMessage).To(ContainSubstring(connectorPod.Pod.Status.PodIP))
```

Comment:
I think this verification can also be contained in `runConnectivityTest`  now that you've combined the data exchange and source IP preservation checks into one spec.  The equivalent of the prior `TestPod2ServiceTCP` is needed by other tests we're adding.

Reply:
Sure why not